src/search/pdbs/pattern_database.cc

```diff
@@ -258,8 +282,75 @@ void PatternDatabase::create_pdb(
             if (alternative_cost < distances[predecessor]) {
                 distances[predecessor] = alternative_cost;
                 pq.push(alternative_cost, predecessor);
+                if (compute_plan) {
+                    generating_op_ids[predecessor] = op_id;
+                    if (op.get_cost() == 0) {
+                        number_of_zero_cost_ops[predecessor] = number_of_zero_cost_ops[state_index] + 1;
+                    } else {
+                        number_of_zero_cost_ops[predecessor] = 0;
+                    }
+                }
+            } else if (alternative_cost == distances[predecessor] && compute_plan && op.get_cost() == 0) {
+                int pred_num_zero_cost_ops = number_of_zero_cost_ops[predecessor];
+                int num_zero_cost_ops_on_current_path = number_of_zero_cost_ops[state_index] + 1;
+                if (num_zero_cost_ops_on_current_path < pred_num_zero_cost_ops) {
+                    generating_op_ids[predecessor] = op_id;
+                    number_of_zero_cost_ops[predecessor] = num_zero_cost_ops_on_current_path;
+                }
+            }
+        }
+    }
+
+    // Compute abstract plan
+    if (compute_plan) {
```

Comment:
This is a long function that does many things. There is a general guideline that a comment like the above "Compute abstract plan" in front of a larger block of code usually means that this block wants to be its own function. But this is not always easy, and I don't want to keep things up with little polishing suggestions like this. Just food for thought.

Reply:
You are absolutely right. I added this within `create_pdb` to mimic the other steps, which are also not separate function, although this is a very stupid argument admittedly :) In the patch where I started separating the computation of PDBs from PatternDatabase, all of the blocks in this method here (`create_pdb`) have become separate steps. But this requires passing around a lot things. I'll leave this as it is for the moment because we might need to address the separation of the PDB computation anyway if Jendrik is right about a possible performance issue due to the added members in this class.